go/store/prolly/tree_diff_test.go

```diff
@@ -0,0 +1,206 @@
+// Copyright 2021 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package prolly
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"sort"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/dolthub/dolt/go/store/val"
+)
+
+func TestMapDiff(t *testing.T) {
+	scales := []int{
+		10,
+		100,
+		1000,
+		10000,
+	}
+
+	for _, s := range scales {
+		name := fmt.Sprintf("test proCur map at scale %d", s)
+		t.Run(name, func(t *testing.T) {
+			prollyMap, tuples := makeProllyMap(t, s)
+			require.Equal(t, s, len(tuples))
+
+			t.Run("empty map diff", func(t *testing.T) {
```

Comment:
Nit: this seems more like "equal map diff"...agreed the set of diffs is empty, but it sounds like we're diffing an empty map.

Might be good to add: diff empty vs. prollyMap; diff prollyMap vs. empty; diff where the callback returns an error forwards the error along.